numpy/lib/function_base.py

```diff
@@ -1501,14 +1509,16 @@ def unwrap(p, discont=pi, axis=-1):
     array([ 0.        ,  0.78539816,  1.57079633, -0.78539816,  0.        ]) # may vary
 
     """
+    if discont is None:
+        discont = (max_val - min_val)/2
     p = asarray(p)
     nd = p.ndim
     dd = diff(p, axis=axis)
     slice1 = [slice(None, None)]*nd     # full slices
     slice1[axis] = slice(1, None)
     slice1 = tuple(slice1)
-    ddmod = mod(dd + pi, 2*pi) - pi
-    _nx.copyto(ddmod, pi, where=(ddmod == -pi) & (dd > 0))
+    ddmod = mod(dd - min_val, max_val - min_val) + min_val
+    _nx.copyto(ddmod, max_val, where=(ddmod == min_val) & (dd > 0))
```

Comment:
Any idea what this `ddmod == min_val` condition is for? Can you produce a case that changes behavior based on whether this condition is here?

Reply:
Not sure, I guess it might come up more with radian sequences and things actually equalling `pi`?

Reply:
Probably easier to investigate only the integer case, floating point rounding is just a distraction.